src/grpc_routes.py

```diff
@@ -31,27 +31,31 @@ def startThread(self):
         asyncio.get_event_loop().run_until_complete(self.run())
 
     async def run(self):
-        async with grpc.aio.insecure_channel(self.address) as channel:
-            self.stub = RoutesStub(channel)
-
-            # Registering antenna
-            self.antenna_id = await self.register_antenna(self.coordinate_x, self.coordinate_y)
-            if self.antenna_id < 0:
-                raise Exception("Failed to register antenna")
-            
-            print(f'Registered antenna with id: {self.antenna_id}')
-
-            while True:
-                message: Message = self.dataQueue.get(True)
-
-                requestMessage: LogMeasurementRequest = LogMeasurementRequest(
-                    aid = self.antenna_id,
-                    identifier = message.identifier,
-                    timestamp = message.timestamp,
-                    signal_strength = message.signal_strength
-                )
-                await self.log_measurement(requestMessage)
-                self.dataQueue.task_done()
+        try:
+            async with grpc.aio.insecure_channel(self.address) as channel:
+                self.stub = RoutesStub(channel)
+
+                # Registering antenna
+                self.antenna_id = await self.register_antenna(self.coordinate_x, self.coordinate_y)
+                if self.antenna_id < 0:
+                    raise Exception("Failed to register antenna")
+                
+                print(f'Registered antenna with id: {self.antenna_id}')
+
+                while True:
+                    message: Message = self.dataQueue.get(True)
+
+                    requestMessage: LogMeasurementRequest = LogMeasurementRequest(
+                        aid = self.antenna_id,
+                        identifier = message.identifier,
+                        timestamp = message.timestamp,
+                        signal_strength = message.signal_strength
+                    )
+                    await self.log_measurement(requestMessage)
+                    self.dataQueue.task_done()
+        except:
+            self.run()
```

Comment:
dummy dummy forgot timeout

Reply:
haha its quite funny that he forgor that. haha

Reply:
HAHAHA This man does not have any programing socks HAHAHAHHA